File: BackEnd/src/domain/grade.ts
import { GradeId, SchoolId, UserId } from './ids';

export enum GradeName {
  First = 'First Grade',
  Second = 'Second Grade',
  Third = 'Third Grade',
}
export class Grade {
  gradeId: GradeId;
  gradeName: GradeName; //chnge this
  gradeSection?: string | null;
  schoolName?: string | null;
  teacherId: UserId;
}

File: Frontend/GP/Models/Grade.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class Grade : BaseEntity
    {
        [Key]
        public int GradeId { get; set; }

            public string GradeName { get; set; } // change this

            public string? SchoolName { get; set; }

            public int TeacherId { get; set; }
            public  Teacher Teacher { get; set; }   
    }
}
